packages/react-storage/src/components/StorageBrowser/adapters/createAmplifyAuthAdapter.ts

```diff
@@ -60,6 +61,13 @@ export const createAmplifyAuthAdapter = (input?: {
     getLocationCredentials,
     listLocations,
     region,
-    registerAuthListener: () => null,
+    registerAuthListener: (onStateChange) => {
+      const cancel = Hub.listen('auth', (data) => {
+        if (data.payload.event === 'signedOut') {
+          onStateChange();
+          cancel();
+        }
+      });
```

Comment:
Nit: prefer `remove` over `cancel` to align to event listener common naming, e.g. `removeEventListener`